src/share/poudriere/jail.sh

```diff
@@ -125,6 +129,28 @@ delete_jail() {
 	cache_dir="${POUDRIERE_DATA}/cache/${JAILNAME}-*"
 	rm -rf ${POUDRIERED}/jails/${JAILNAME} ${cache_dir} || :
 	echo " done"
+	if [ "${CLEAN}" == "none" ]; then
+		return 0
+	fi
+	msg_n "Cleaning ${JAILNAME} data..."
+	case ${CLEAN} in
+		all)
+			find ${POUDRIERE_DATA} -name "${JAILNAME}" -type d | xargs rm -rf || :
+			;;
+		logs)
+			logs_dir="${POUDRIERE_DATA}/logs/bulk/${JAILNAME}-*"
+			rm -rf ${logs_dir} || :
+			;;
+		packages)
+			pkgs_dir="${POUDRIERE_DATA}/packages/${JAILNAME}-*"
+			rm -rf ${pkgs_dir} || :
+			;;
+		wrkdirs)
+			wrk_dir="${POUDRIERE_DATA}/wrkdirs/${JAILNAME}*"
+			rm -rf ${wrk_dir} || :
+			;;
+	esac
+	echo " done"
 }
```

Comment:
You should also add a clean option of _cache_ to remove `${POUDRIERE_DATA}/cache/JAILNAME-*` (`finddir="${POUDRIERE_DATA}/cache"; depth=1`)